src/hdmf/data_utils.py

```diff
@@ -200,13 +221,19 @@ def _read_next_chunk(self):
             else:
                 self.__next_chunk.data = np.asarray(curr_next_chunk)
                 if self.__next_chunk.selection is None:
-                    self.__next_chunk.selection = slice(0, next_chunk_size)
+                    self.__next_chunk.selection = slice(start=curr_chunk_offset,
+                                                        stop=curr_chunk_offset+next_chunk_size)
```

Comment:
@rly the changes in line 204 to 225 should hopefully do the trick to skip blocks